.markdownlint.json

```diff
@@ -0,0 +1,7 @@
+{
```

Comment:
For 24 we should look into tabs part if there is way to do it without h2.
Only MD033 should be here (that is also questionable).
Rest should not be disabled.

Reply:
We can see this as a first pass and improve later.

* MD013 (line length) - it really seems like a stylistic thing. I personally prefer writing paragraphs with no line breaks (I think most of our docs are like this).
* MD024 (double heading) - we just discussed this in the other issue (CATS). FWIW I agree we shouldn't disable this, but for now it seems necessary for the tabs. I suggest we leave it disabled for now and look into the tabs markdown separately.
* MD025 is a technicality - the linter sees the title in the YAML front matter and complains that our H1 is a duplicate - but the doc system requires it, we instruct it to disregard the YAML. FWIW [the docs explicitly talk about this option](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md025)).
* MD028 (line break between block quotes) - I don't get this rule, in some cases we just need to notes or warnings with nothing in between. We don't want to merge them, and we don't have anything to say between them. Simple.
* MD033 (allow HTML elements - AFAIK there's no way to do certain things in pure markdown (e.g. superscript).

Reply:
- I don't see line length warning for paragraphs. What are you referring to?
- Double heading, tabs thing we know so we can ignore in code. No need to disable for entire docs where something can go amiss.
- I am not seeing MD025 warnings either. Can you elaborate more?
- We should defer on specific case.

I don't see this as good first pass. Where exactly they are causing errors and why do we need to suppress that blanketly. It certainly isn't in build warnings.

Reply:
> I don't see line length warning for paragraphs. What are you referring to?

[MD013](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md013---line-length) is about setting maximum length (default by 80). Unless you suppress it, you have to manually put line breaks on each line before 80. I'm using VS Code's markdownlint plugin which generates warnings for this, maybe your plugin or linter doesn't?

> Double heading, tabs thing we know so we can ignore in code. No need to disable for entire docs where something can go amiss.

Fine.

> I am not seeing MD025 warnings either. Can you elaborate more?

Sure. A markdown file is supposed to have only one top-level heading (for the whole page) - in all our pages we have one. The problem is that the linter *also* recognizes the `title` key in the YAML front matter, and assumes that whoever's parsing/generating the document makes that into an <H1> in the resulting HTML - this unfortunately isn't the case in our doc system. So the linter thinks we have two top-level headings: one from the YAML, another from the title we manually put on each document.

Note that I didn't disable this rule - I just configured it not to recognize the `title` key in the YAML front matter.

> We should defer on specific case.

Agreed.

> I don't see this as good first pass. Where exactly they are causing errors and why do we need to suppress that blanketly. It certainly isn't in build warnings.

Well, considering there was nothing before, I'd say it's at least a start - I'd rather merge and discuss contention points later. Let's not forget - this is about linting our doc markdown sources :)

The CI pipeline apparently doesn't lint markdown (unfortunately). I'm using VS Code with a markdown linter which generates squigglies. There's also a cmdline tool

Reply:
> Agreed.

Remove MD028 then.

Reply:
> Remove MD028 then.

Reply:
As I wrote above, this would make it impossible to have one block note after the other, like the two tips in the [disconnected entities page](https://docs.microsoft.com/en-us/ef/core/saving/disconnected-entities). If you remove the space between them as the linter wants, they get merged into a single note. AFAIK the note syntax is docfx-specific, which is probably the linter's rule doesn't work well here.

Reply:
Disable in that specific case it is possible to suppress in specific place.
I asked to defer for MD28, you wrote agreed then why....

Reply:
Sorry, I had an off-by-one and thought your comment was on MD033 not MD028. Done.

Reply:
MD033 is totally fine. We can expand that list as we find more html tags which we need in our syntax.

Reply:
OK. Removed MD028 suppression so I think this should be good to merge, can someone approve?

Reply:
Think you want a `"default": true` at the top like https://github.com/dotnet/docs/blob/master/.markdownlint.json#L2

Reply:
Thanks @nsconni, done.